p22/NOTES.md

```diff
@@ -0,0 +1,11 @@
+# Links
+
+## documentation:
+
+- Article: https://www.tangramvision.com/blog/making-great-docs-with-rustdoc#-arguments
+- Example for good documentation: https://doc.rust-lang.org/src/alloc/vec/mod.rs.html#944
+
+## Memoization
+
+TODO: Find a good solution for the recursive version.
```

Comment:
You recursive function looks good to me. You have the intended solution.

Reply:
the comment is mostly about memoization and how to reuse fib(n-2) common to fib(n-1)=>fib(n-2) + fib(n-3) and fib(n-2).

Reply:
Memoization is not needed here, as one of the goals of exercise is to have "slow" implementation of Fibonacci function to see the difference between it and the loop implementation in the benchamrks.

Reply:
ok